tests/TestHelpers/InbucketHelper.php

```diff
@@ -0,0 +1,208 @@
+	 * A mailbox is created automatically in InBucket for every unique email sender|receiver.
+	 *
+	 * @param string $mailBox
+	 * @param string|null $xRequestId
+	 *
+	 * @return mixed JSON encoded contents
+	 * @throws GuzzleException
+	 */
+	public static function getMailboxInformation(string $mailBox, ?string $xRequestId = null) {
+		$response = HttpRequestHelper::get(
+			self::getLocalEmailUrl() . "/api/v1/mailbox/" . $mailBox,
+			$xRequestId,
+			null,
+			null,
+			['Content-Type' => 'application/json']
+		);
+		return json_decode($response->getBody()->getContents());
+	}
+
+	/**
+	 * Deletes all the emails for the provided mailbox
+	 *
+	 * @param string $localInbucketUrl
+	 * @param string|null $xRequestId
+	 * @param string $mailBox
+	 *
+	 * @return ResponseInterface
+	 * @throws GuzzleException
+	 */
+	public static function deleteAllEmailsForAMailbox(
+		string $localInbucketUrl,
+		?string $xRequestId,
+		string $mailBox
+	):ResponseInterface {
+		return HttpRequestHelper::delete(
+			$localInbucketUrl . "/api/v1/mailbox/" . $mailBox,
+			$xRequestId
+		);
+	}
+
+	/**
+	 * Returns the body of the last received email for the provided receiver according to the provided email address and the serial number
+	 * For email number, 1 means the latest one
+	 *
+	 * @param string $emailAddress
+	 * @param string|null $xRequestId
+	 * @param int|null $emailNumber For email number, 1 means the latest one
+	 * @param int|null $waitTimeSec Time to wait for the email if the email has been delivered
+	 *
+	 * @return string
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public static function getBodyOfLastEmail(
+		string $emailAddress,
+		string $xRequestId,
+		?int $emailNumber = 1,
+		?int $waitTimeSec = EMAIL_WAIT_TIMEOUT_SEC
+	): string {
+		$currentTime = \time();
+		$endTime = $currentTime + $waitTimeSec;
+		$mailBox = self::getMailBoxFromEmail($emailAddress);
+		while ($currentTime <= $endTime) {
+			$mailboxResponse = self::getMailboxInformation($mailBox);
+			if (!empty($mailboxResponse) && \sizeof($mailboxResponse) >= $emailNumber) {
+				$mailboxId = $mailboxResponse[\sizeof($mailboxResponse) - $emailNumber]->id;
+				$response = self::getContentOfAnEmailByID($mailBox, $mailboxId, $xRequestId);
+				$body = \str_replace(
+					"\r\n",
+					"\n",
+					\quoted_printable_decode($response->body->text . "\n" . $response->body->html)
+				);
+				return $body;
+			}
+			\usleep(STANDARD_SLEEP_TIME_MICROSEC * 50);
+			$currentTime = \time();
+		}
+		throw new Exception("Could not find the email to the address: " . $emailAddress);
+	}
+
+	/**
+	 * returns body content of a specific email (mailBox) with email ID (mailbox Id)
+	 *
+	 * @param string $mailBox
+	 * @param string $mailboxId
+	 * @param string|null $xRequestId
+	 *
+	 * @return mixed JSON encoded contents
+	 * @throws GuzzleException
+	 */
+	public static function getContentOfAnEmailById(string $mailBox, string $mailboxId, ?string $xRequestId = null) {
```

Comment:
```suggestion
	public static function getBodyOfAnEmailById(string $mailBox, string $mailboxId, ?string $xRequestId = null) {
```
just making the naming a bit more consistent because you have called the other function `getBodyOfLastEmail`